### ISSUE TITLE
Some warnings when running typecheck manually

### ISSUE THREAD

**Issue by taylorthurlow** [castwide/solargraph#198]
Here's my console output:

~~~
$ solargraph typecheck app/models/order_change.rb 

[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/selenium-webdriver-3.142.3/selenium-webdriver.gemspec: [RuntimeError] cwd must be /my/repo/vendor/ruby/2.5.0/gems/selenium-webdriver-3.142.3 when reading gemspec
[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/crass-1.0.4/crass.gemspec: [LoadError] cannot load such file -- ./lib/crass/version
[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/raindrops-0.19.0/raindrops.gemspec: [Errno::ENOENT] No such file or directory @ rb_sysopen - README
fatal: not a git repository: 'ext/ffi_c/libffi/.git'
fatal: not a git repository: 'ext/ffi_c/libffi/.git'
[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/kgio-2.11.2/kgio.gemspec: [Errno::ENOENT] No such file or directory @ rb_sysopen - .manifest
[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/rmagick-3.1.0/rmagick.gemspec: [LoadError] cannot load such file -- ./lib/rmagick/version
[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/unicorn-4.8.2/unicorn.gemspec: [Errno::ENOENT] No such file or directory @ rb_sysopen - .manifest
[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/json-1.8.6/json-java.gemspec: [Errno::ENOENT] No such file or directory @ rb_sysopen - VERSION
[WARN] Error reading /my/repo/vendor/ruby/2.5.0/gems/rack-2.0.7/rack.gemspec: [Errno::ENOENT] No such file or directory @ rb_sysopen - lib/rack.rb
/my/repo/app/models/order_change.rb:10 - get_name_by_id has undefined @param type for id
/my/repo/app/models/order_change.rb:10 - get_name_by_id has undefined @return type
/my/repo/app/models/order_change.rb:14 - name has undefined @return type
/my/repo/app/models/order_change.rb:35 - validate_change has undefined @return type
4 problems found.
~~~

Not sure if the `not a git repository` and other warnings are related.

**Comment by castwide** [comment]
The workspace mapping process collects information about your workspace from its .gemspec files. If it encounters an exception reading them, it logs the error and keeps going.

In this particular case, the gemspecs are in vendored gems, so they should probably be ignored. I'll look into fixing it.

**Comment by castwide** [comment]
There's a fix for this in the `master` branch. It'll be in the next gem release.

**Comment by castwide** [comment]
Fixed in 0.34.0.
